### PR TITLE
libYARP_OS-YarpPlugin: change Time::now() with SystemClock::nowSystem()

### LINKED ISSUE #1265: Module does not start when defining YARP_CLOCK
I am trying to test a module I have written using gazebo, but if I append ``YARP_CLOCK=/clock`` before launching it, the module get stuck after the following message
```
yarp: Anonymous port active at tcp://10.255.111.71:38400/
```
If I launch it without the clock port it starts smoothly.

### PR DESCRIPTION
**Pull request by Nicogene**
This PR fixes #1265.

Please review code.

### CODE REVIEW

File: src/libYARP_OS/src/YarpPlugin.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #include <yarp/os/ResourceFinder.h>
 #include <yarp/os/Time.h>
```
Comment (francesco-romano) [inline src/libYARP_OS/src/YarpPlugin.cpp]: We can probably remove this header as it is no longer necessary

**Comment by francesco-romano** [comment]
A part from the minor comment on the header, it looks good to me.
Thanks @Nicogene

**Comment by francesco-romano** [comment]
@Nicogene I tested also on a more complex example and it works.
For me it can be merged as soon as CIs are happy

**Comment by drdanz** [comment]
Merged, thanks

**Pull request merged**